packages/react-native-codegen/src/parsers/parsers-commons.js

```diff
@@ -100,10 +105,61 @@ function emitMixedTypeAnnotation(
   });
 }
 
+function remapUnionTypeAnnotationMemberNames(
+  types: $FlowFixMe,
+  language: ParserType,
+): UnionTypeAnnotationMemberType[] {
+  const remapLiteral = (item: $FlowFixMe) => {
+    if (language === 'Flow') {
+      return item.type
+        .replace('NumberLiteralTypeAnnotation', 'NumberTypeAnnotation')
+        .replace('StringLiteralTypeAnnotation', 'StringTypeAnnotation');
+    }
+
+    return item.literal
+      ? item.literal.type
+          .replace('NumericLiteral', 'NumberTypeAnnotation')
+          .replace('StringLiteral', 'StringTypeAnnotation')
+      : 'ObjectTypeAnnotation';
+  };
+
+  return types
+    .map(remapLiteral)
+    .filter((value, index, self) => self.indexOf(value) === index);
```

Comment:
If I'm not wrong, after the `map` invocation we have a list of strings, with the types of the union. The logic of this filter is a little hard to understand: it checks that, for each value, the index in which it is is exactly the same index of the iteration. If yes, it returns it, if not, it doesn't. So basically, we are removing duplicates.

What if we use a Set for this?
```suggestion
  return new Set(types
    .map(remapLiteral))
```

I would also update the return type.

Reply:
Thanks for the review, it's done !